### PR TITLE
rustwasm bookshelf

### PR DESCRIPTION
**Pull request by DebugSteven**
This PR fleshes out the `docs.md` file to provide some
context about what the Rust and WebAssembly resources are
and what the reader might want to know before jumping into
these books. This PR is a continuation of @alexcrichton's PR
to [merge the documentation for Rust and WebAssembly to one
location](https://github.com/rustwasm/rustwasm.github.io/pull/37).

The `docs.md` format is inspired by ["The Rust Bookshelf"](https://github.com/rust-lang/rust/blob/master/src/doc/index.md).
Once this bookshelf is merged, it might be nice to send the
docs team a PR to add a link to the Rust and WebAssembly
Bookshelf under the Specialize Rust section so that more
people can find this information in the general docs too!
I'd be happy to send that PR if that's something that would
be useful. I'd would appreciate feedback on what the
Rust and WebAssembly section should say. The Embedded
Working Group is under the Specialize Rust section
currently, if you'd like to look at an example.

Tentatively, I think this would work well as a description
to link to the Rust and WebAssembly Working Group and
the bookshelf:
```markdown
## WebAssembly

To use Rust in your web browser or to interop with JavaScript, check out the resources maintained by
the [WebAssembly Working Group].

[WebAssembly Working Group]: https://github.com/rustwasm

### The Rust and WebAssembly Bookshelf

[The Rust and WebAssembly Bookshelf] contains resources for those just getting started with Rust and WebAssembly to those that would like a deeper dive into the various tools and crates in the rustwasm project.

[The Rust and WebAssembly Bookshelf]: https://rustwasm.github.io/docs.html
```

### CODE REVIEW

File: docs.md [hunk 1]
```diff
@@ -1,3 +1,81 @@
...
+#search-but:hover, #search-input:focus {
+    border-color: #55a9ff;
+}
+h2 {
+    font-size: 18px;
+}
+</style>
+
+Welcome to an overview of the documentation provided by the [Rust
+WebAssembly Working Group](https://github.com/rustwasm). All of these
```
Comment (fitzgen) [inline docs.md]:
```suggestion
and WebAssembly Working Group](https://github.com/rustwasm). All of these
```

File: docs.md [hunk 2]
```diff
@@ -1,3 +1,81 @@
...
+projects are managed by the working group; there are other unofficial
+documentation resources as well!
+
+Many of these resources take the form of "books"; we collectively call
+these "The Rust WebAssembly Bookshelf." Some are large, some are small.
```
Comment (fitzgen) [inline docs.md]:
```suggestion
these "The Rust and WebAssembly Bookshelf." Some are large, some are small.
```

File: docs.md [hunk 3]
```diff
@@ -1,3 +1,81 @@
...
+of Rust and that you have some familiarity with JavaScript. If you want
+to learn how to use Rust in your web browser, these resources are a
+great place to get started:
+
+## The Rust Wasm Book
+
+[The Rust Wasm Book](docs/book) describes how to use Rust and
+WebAssembly together. There's a tutorial to create an end to end project
+using Rust and WebAssembly and finally the book concludes with reference
+sections full of general, good to know, information, to explore more
```
Comment (fitzgen) [inline docs.md]:
```suggestion
sections full of general, good-to-know information to explore more
```

**Comment by fitzgen** [comment]
Will merge as soon as the inline suggestions are addressed :)

**Comment by fitzgen** [comment]
> I'd be happy to send that PR if that's something that would
> be useful.

+1000

> I'd would appreciate feedback on what the
> Rust and WebAssembly section should say.

What you've written in the original comment sounds great to me. The only nitpick I have is that I'd use the full "interoperate" instead of "interop" :-p

**Comment by DebugSteven** [comment]
Thank you for the review on this! 
Also it's only Tuesday, I'd say that's pretty fast turn around for review. :)

**Comment by alexcrichton** [comment]
As a follow up I've sent a number of PRs to indicate that the gh-pages documentation we're primarily using today is for unpublished versions and the official versions are now in this repository's published documentation:

* https://github.com/rustwasm/wasm-pack/pull/565
* https://github.com/rustwasm/wasm-bindgen/pull/1329
* https://github.com/rustwasm/book/pull/155

**Pull request merged**